src/EditorFeatures/CSharpTest/CodeActions/ConvertToSwitch/ConvertToSwitchTests.cs

```diff
@@ -0,0 +1,378 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ConvertToSwitch;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeActions.ConvertToSwitch
+{
+    public class ConvertToSwitchTests : AbstractCSharpCodeActionTest
+    {
+        protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspace workspace)
+            => new CSharpConvertToSwitchCodeRefactoringProvider();
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertToSwitch)]
+        public async Task TestUnreachableEndPoint()
+        {
+            await TestAsync(
+@"class C
+{
+    void M(int i)
+    {
+        [||]if (i == 1 || i == 2 || i == 3)
+            return;
+    }
+}",
+@"class C
+{
+    void M(int i)
+    {
+        switch(i)
+        {
+            case 1:
+            case 2:
+            case 3:
+                return;
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertToSwitch)]
+        public async Task TestReachableEndPoint()
+        {
+            await TestAsync(
+@"class C
+{
+    void M(int i)
+    {
+        [||]if (i == 1 || i == 2 || i == 3)
+            M(i);
+    }
+}",
+@"class C
+{
+    void M(int i)
+    {
+        switch(i)
+        {
+            case 1:
+            case 2:
+            case 3:
+                M(i);
+                break;
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertToSwitch)]
+        public async Task TestMultipleCases_01()
+        {
+            await TestAsync(
+@"class C
+{
+    void M(int i)
+    {
+        [||]if (i == 1 || 2 == i || i == 3) M(0);
+        else if (i == 4 || 5 == i || i == 6) M(1);
```

Comment:
what if the 'if' and the 'else' declare variables with the same name.  In the switch you could end up with a conflict right, since the cases don't have blocks within then to ensure proper scopes.